lib/posthog/client.rb

```diff
@@ -139,58 +142,56 @@ def queued_messages
       @queue.length
     end
 
-    def is_feature_enabled(flag_key, distinct_id, default_value = false, groups = {})
-      unless @personal_api_key
-        logger.error(
-          'You need to specify a personal_api_key to use feature flags'
-        )
-        return
-      end
-      is_enabled =
-        @feature_flags_poller.is_feature_enabled(
-          flag_key,
-          distinct_id,
-          default_value,
-          groups
-        )
-      capture(
-        {
-          'distinct_id': distinct_id,
-          'event': '$feature_flag_called',
-          'properties': {
-            '$feature_flag': flag_key,
-            '$feature_flag_response': is_enabled
-          }
-        }
-      )
-      return is_enabled
+    def is_feature_enabled(flag_key, distinct_id, default_value = false, groups: {}, person_properties: {}, group_properties: {})
+      return !!get_feature_flag(flag_key, distinct_id, default_value, groups: groups, person_properties: person_properties, group_properties: group_properties)
     end
 
-    def get_feature_flag(key, distinct_id, groups={})
-      unless @personal_api_key
-        logger.error(
-          'You need to specify a personal_api_key to use feature flags'
+    # Returns whether the given feature flag is enabled for the given user or not
+    #
+    # @param [String] key The key of the feature flag
+    # @param [String] distinct_id The distinct id of the user
+    # @param [Hash] groups
+    # @param [Hash] person_properties key-value pairs of properties to associate with the user.
+    # @param [Hash] group_properties
+    # 
+    # @return [String, nil] The value of the feature flag
+    #
+    # The provided properties are used to calculate feature flags locally, if possible.
+    #
+    # `groups` are a mapping from group type to group key. So, if you have a group type of "organization" and a group key of "5",
+    # you would pass groups={"organization": "5"}.
+    # `group_properties` take the format: { group_type_name: { group_properties } }
+    # So, for example, if you have the group type "organization" and the group key "5", with the properties name, and employee count,
+    # you'll send these as:
+    # ```python
```

Comment:
Having python heredocs in ruby code is kind of sus :)

Reply:
heh, nice catch 🙈